flink-core/src/main/java/org/apache/flink/configuration/BlobServerOptions.java

```diff
@@ -102,4 +102,20 @@
 	public static final ConfigOption<Integer> OFFLOAD_MINSIZE = key("blob.offload.minsize")
 		.defaultValue(1_024 * 1_024) // 1MiB by default
 		.withDescription("The minimum size for messages to be offloaded to the BlobServer.");
+
+	/**
+	 * The socket timeout in milliseconds for the blob client.
+	 */
+	public static final ConfigOption<Integer> SO_TIMEOUT =
+		key("blob.client.socket.timeout")
+			.defaultValue(120_000)
```

Comment:
` 2 minutes` is a value that I give freely, and it has little basis, just considering that a slightly longer wait may be required in the case of high CPU load. In fact, the blob client has a retry mechanism. and we should set it a little smaller. Do you have any recommended values about connection and socket timeouts?

Reply:
Check the option `taskmanager.network.netty.client.connectTimeoutSec`, whose default value is set to `120s`, and let we keep connection/socket timeouts consistent with it.  What do you think? @tillrohrmann

Reply:
The reason I'm asking is because we might break existing setups with this default value. Imagine you have a really large setup (let's say 100 TMs) and all of them need to download a large set of blobs. Since the `BlobServer` only serves 50 connections concurrently, it might be the case that 50 others will time out if the transmission takes longer than 2 minutes. I think the Netty case is a bit different because we don't limit the number of concurrent transactions. So concretely, I'm wondering whether we should set the default value to `0` to maintain the existing behaviour or to something a bit larger (e.g. 10 minutes). Or alternatively, we might add a bit more meaningful error message allowing users to pin point the problem and what configuration value to increase.

Reply:
You're right. I didn't think of that.  Or the default connection timeout is set to `0`, while SO_TIMEOUT is still set to `2 minutes` ?

Reply:
I think this could be a good compromise. Then let's set `blob.client.socket.timeout` to `5` minutes and `blob.client.connect.timeout` to `0`.